contracts/src/ethereum/TokenBridge.sol

```diff
@@ -133,16 +137,17 @@ contract TokenBridge is
 
     function sendMessageStaticAToken(address l1Token, uint256 accRewards) 
         external
-        isApprovedToken(l1Token)
     {
       uint256 l2Token = l1TokentoL2Token[l1Token];
 
-      uint256[] memory payload = new uint256[](4);
-      payload[0] = block.number;
-      payload[1] = l2Token;
-      (payload[2], payload[3]) = toSplitUint(accRewards);
+      if (isValidFelt(l2Token)) {
```

Comment:
we should only send an update message when L1 staticAToken has an L2 counterpart. the function `sendMessageStaticAToken` will now be called before every transfer of L1 staticAToken, some of them may only live in L1 or may not have been registered yet on the token bridge.